### DISCUSSION TITLE
Ideas and proposals

### DISCUSSION THREAD

**Discussion by llcc** [amake/orgro#112] [General]
Orgro has gradually become the most elegant and best viewer for Orgmode documents on mobile after extensive polishing,  also, it holds the potential to become one of the top Orgmode editors in the future. For further development of Orgro, I'd like to open this discussion to gather ideas and proposals to make Orgro even better.

Key Improvement Areas:

1. Org Agenda. This feature has been requested in Issue #92. Implementing a functional Org agenda will greatly enhance productivity.
2. Enhance the buffer or section editor. 
3. Powerful Image Editor. I found that editing and annotating images feels more intuitive on mobile, especially with an Apple Pencil. Currently, long-pressing an image opens it in a viewer, we can integrate the [pro_image_editor](https://pub.dev/packages/pro_image_editor) package to provide image annotation functionality for orgro.
4. Performance Optimization metioned in Issue #52. I also feel the pain of noticed lag when scrolling, cycling through headlines or documents, or searching in the Sparse Tree. As @amake mentioned, this is a long-term challenge. However, in the short term, adding a performance metrics UI could help gather bottleneck data for further optimization.
5. Ecosystem and Plugins. In the long run, Orgro could benefit from a plugin system to allow third-party or user-defined code to connect with the app. This could address suggestions like those in Issue #87 and Issue #97 if the community loves to contribute. For example, an in-app PDF viewer using packages like  ([flutter_cached_pdfview](https://pub.dev/packages/flutter_cached_pdfview) or [flutter_pdfview](https://pub.dev/packages/flutter_pdfview) ) can be implemented. Developing more plugins would expand Orgro’s functionality even further.

More ideas and proposals are welcomed.

**Comment by llcc** [comment]
Another one:
The following Org meta line should be treated as same as table controller.

`#+call: `, `#+caption:` , `#+tblfm:`

**Reply by amake** [reply]
I'm not sure what you mean by "same as table controller". Can you clarify?

**Reply by llcc** [reply]
The above meta lines should be scrollable horizontally, similar to the table. Specifically, the `#+tlbfm` line, which belongs to the table, would be better to scroll together with the table.

**Reply by amake** [reply]
In v1.46.2 meta lines will scroll horizontally in Reader Mode. `#+tlbfm` won't get special treatment yet.

**Comment by llcc** [comment]
Great if https://github.com/nobiot/org-transclusion can be implemented.

**Comment by llcc** [comment]
[Org property inheritance](https://orgmode.org/manual/Property-Inheritance.html) is not supported yet. 

The following feature is also not supported.
```org
If you want to add to the value of an existing property, append a ‘+’ to the property name. The following results in the property ‘var’ having the value ‘foo=1 bar=2’.

#+PROPERTY: var  foo=1
#+PROPERTY: var+ bar=2 
```

**Comment by kresdjan** [comment]
Font suggestion. Add Cascadia Code to list of fonts. https://github.com/microsoft/cascadia-code

**Reply by amake** [reply]
This will be available in v1.46.0, ready for testing now:

- [iOS](https://testflight.apple.com/join/8YTxcuZC)
- [Android](https://play.google.com/store/apps/details?id=com.madlonkay.orgro)

**Comment by bbigras** [comment]
Support for https://orgmode.org/manual/Initial-visibility.html (like `#+STARTUP: show2levels`)

**Reply by amake** [reply]
This is already supported (though the `show[2..5]levels` options are treated as aliases for `content`).

https://github.com/amake/orgro/blob/master/assets/manual/orgro-manual.org#startup-keywords

**Comment by bbigras** [comment]
Support for custom `org-todo-keywords`.

I have customs `org-todo-keywords` like `PROJ` that are not currently highlighted in orgro.

**Reply by amake** [reply]
This is already supported: https://github.com/amake/orgro/blob/master/assets/manual/orgro-manual.org#todo-keywords
